src/datasources/notifications/notifications.datasource.spec.ts

```diff
@@ -0,0 +1,633 @@
+        ]);
+        expect(types).toStrictEqual(
+          Object.values(NotificationType).map((type) => {
+            return {
+              id: expect.any(Number),
+              name: type,
+            };
+          }),
+        );
+        expect(subscriptions).toStrictEqual(
+          upsertSubscriptionsDto.safes
+            .map((safe, i) => {
+              return {
+                id: i + 1,
+                signer_address: signerAddress,
+                push_notification_device_id: devices[0].id,
+                chain_id: safe.chainId,
+                safe_address: safe.address,
+                created_at: expect.any(Date),
+                updated_at: expect.any(Date),
+              };
+            })
+            .concat(
+              secondUpsertSubscriptionsDto.safes.map((safe, i) => {
+                return {
+                  id: upsertSubscriptionsDto.safes.length + i + 1,
+                  signer_address: signerAddress,
+                  push_notification_device_id: devices[1].id,
+                  chain_id: safe.chainId,
+                  safe_address: safe.address,
+                  created_at: expect.any(Date),
+                  updated_at: expect.any(Date),
+                };
+              }),
+            ),
+        );
+        expect(subscribedNotifications).toStrictEqual(
+          expect.arrayContaining(
+            upsertSubscriptionsDto.safes
+              .flatMap((safe, i) => {
+                return safe.notificationTypes.map((type) => {
+                  return {
+                    id: expect.any(Number),
+                    notification_subscription_id: i + 1,
+                    notification_type_id: types.find((t) => t.name === type)
+                      ?.id,
+                  };
+                });
+              })
+              .concat(
+                secondUpsertSubscriptionsDto.safes.flatMap((safe, i) => {
+                  return safe.notificationTypes.map((type) => {
+                    return {
+                      id: expect.any(Number),
+                      notification_subscription_id:
+                        upsertSubscriptionsDto.safes.length + i + 1,
+                      notification_type_id: types.find((t) => t.name === type)
+                        ?.id,
+                    };
+                  });
+                }),
+              ),
+          ),
+        );
+      });
+    });
+
+    it('should allow multiple subscriptions, varying by signer', async () => {
+      const signerAddress = getAddress(faker.finance.ethereumAddress());
+      const upsertSubscriptionsDto = upsertSubscriptionsDtoBuilder().build();
+      const secondSignerAddress = getAddress(faker.finance.ethereumAddress());
+      await target.upsertSubscriptions({
+        signerAddress,
+        upsertSubscriptionsDto,
+      });
+      await target.upsertSubscriptions({
+        signerAddress: secondSignerAddress,
+        upsertSubscriptionsDto: upsertSubscriptionsDto,
```

Comment:
```suggestion
        upsertSubscriptionsDto,
```

Reply:
Changed in d49aca2e6a097d8ac438fa5876d39f1b2006bf3f.